lib3h/src/real_engine.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::VecDeque;
+
+use holochain_lib3h_protocol::{
+    network_engine::{DidWork, NetworkEngine},
+    protocol::Lib3hProtocol,
+    Lib3hResult,
+};
+
+/// Struct holding all config settings for the RealEngine
+#[derive(Debug, Clone, PartialEq)]
+pub struct RealEngineConfig {
+    pub socket_type: String,
+    pub bootstrap_nodes: Vec<String>,
+    pub work_dir: String,
+    pub log_level: char,
+}
+
+/// Lib3h's 'real mode' as a NetworkEngine
+pub struct RealEngine {
+    /// Config settings
+    config: RealEngineConfig,
+    /// FIFO of messages received from Core
+    inbox: VecDeque<Lib3hProtocol>,
+    /// Identifier
+    name: String,
+}
+
+impl RealEngine {
+    /// Constructor
+    pub fn new(config: RealEngineConfig, name: &str) -> Lib3hResult<Self> {
+        Ok(RealEngine {
+            config,
+            inbox: VecDeque::new(),
```

Comment:
Do we have any idea of the size of this queue ? Maybe we could use [VecDeque::with_capacity(usize)](https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.with_capacity) in order to help the compiler reserving some space on the heap. It's not absolutely required, just throwing some thoughts around :)

Reply:
Don't know the size we want yet, or if we even want to hard limit it.
Once we do, I'm all for reserving the memory space upfront 👍

Reply:
It does limit anything, it just reserves the minimum amount we think we might need. And if it happens to be too small, the allocated size on the head will grow according to our need. But I see your point don't worry, and I think it's better to let it this way, so the compiler can optimize what it can better than us at the moment.